WALKTHROUGH.md

```diff
@@ -85,7 +88,81 @@ A lightweight guide on how to use this tool in context.
     script: AzureSignTool sign -du "[YOUR_URL]" -kvu "https://[VAULT_ID].vault.azure.net" -kvi "[REDACTED_APPLICATION_ID]" -kvt "[REDACTED_DIRECTORY_ID]" -kvs "[REDACTED_APPLICATION_CLIENT_SECRET]" -kvc "[REDACTED_CERT_NAME]" -v [FILES_YOU_WANT_TO_SIGN]
 ```
 
-At this point, the build should be able to run and sign the files you have listed.
+## Using AzureSignTool to sign files with GitLab CI
+
+### A) Using service principal authentication
+
+* Go to KeyVault's the `Access Policies` section.
+* Create an access policy that applies to your service principal.
+* For the access policy, set the below permissions:
+
+  | Area | Permissions |
+  | ---- | ----------- |
+  | Key | Verify, Sign, Get, List |
+  | Secret | Get, List |
+  | Certificate | Get, List |
+
+* In your GitLab build configuration, add a job to install and execute the AzureSignTool:
+
+```yml
+sign:
+  stage: deploy
+  image: mcr.microsoft.com/dotnet/sdk:8.0-windowsservercore-ltsc2019 # If docker on windows is used.
+  before_script:
+    - Invoke-WebRequest https://github.com/vcsjones/AzureSignTool/releases/latest/AzureSignTool-arm64.exe -OutFile AzureSignTool.exe
```

Comment:
> AzureSignTool-arm64.exe

This does not seem like the right download. Should it be `AzureSignTool-x64.exe`? As far as I know Windows Server 2019 does not support ARM64.

Reply:
That's right, I copied it from the README and forgot to change it.